[TEST] Rollup cleanup failing BWC tests on 6.x

Relates to #34342

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+default-distro/4/testReport/junit/org.elasticsearch.backwards/IndexingIT/testIndexVersionPropagation/

Reproduce with:

```
./gradlew :qa:mixed-cluster:v5.6.13-SNAPSHOT#mixedClusterTestRunner -Dtests.seed=8A390CB44222626C -Dtestsecurity.manager=true -Dtests.locale=el -Dtests.timezone=Pacific/Apia -Dtests.distribution=zip -Dcompiler.java=11 -Druntime.java=8
```

```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:38073], URI [/_xpack/rollup/job/_all], status line [HTTP/1.1 400 Bad Request]
No handler found for uri [/_xpack/rollup/job/_all] and method [GET]
	at __randomizedtesting.SeedInfo.seed([8A390CB44222626C:BAA5894E7D4A433]:0)
	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:926)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:229)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeRollupJobs(ESRestTestCase.java:395)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:329)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:179)
	at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:38073], URI [/_xpack/rollup/job/_all], status line [HTTP/1.1 400 Bad Request]
No handler found for uri [/_xpack/rollup/job/_all] and method [GET]
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:546)
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:531)
	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119)
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177)
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436)
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326)
	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
	... 1 more
```

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
@@ -25,6 +25,7 @@ import org.apache.http.message.BasicHeader;
 import org.apache.http.nio.conn.ssl.SSLIOSessionStrategy;
 import org.apache.http.ssl.SSLContexts;
 import org.apache.http.util.EntityUtils;
+import org.elasticsearch.Version;
 import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksAction;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.Response;
@@ -69,6 +70,7 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.TreeSet;
 import java.util.concurrent.TimeUnit;
 
 import static java.util.Collections.sort;
@@ -103,22 +105,10 @@ public abstract class ESRestTestCase extends ESTestCase {
      * Does any node in the cluster being tested have x-pack installed?
      */
     public static boolean hasXPack() throws IOException {
-        RestClient client = adminClient();
-        if (client == null) {
+        if (hasXPack == null) {
             throw new IllegalStateException("must be called inside of a rest test case test");
         }
-        Map<?, ?> response = entityAsMap(client.performRequest(new Request("GET", "_nodes/plugins")));
-        Map<?, ?> nodes = (Map<?, ?>) response.get("nodes");
-        for (Map.Entry<?, ?> node : nodes.entrySet()) {
-            Map<?, ?> nodeInfo = (Map<?, ?>) node.getValue();
-            for (Object module: (List<?>) nodeInfo.get("modules")) {
-                Map<?, ?> moduleInfo = (Map<?, ?>) module;
-                if (moduleInfo.get("name").toString().startsWith("x-pack-")) {
-                    return true;
-                }
-            }
-        }
-        return false;
+        return hasXPack;
     }
 
     private static List<HttpHost> clusterHosts;
@@ -131,12 +121,16 @@ public abstract class ESRestTestCase extends ESTestCase {
      * completes
      */
     private static RestClient adminClient;
+    private static Boolean hasXPack;
+    private static TreeSet<Version> nodeVersions;
 
     @Before
     public void initClient() throws IOException {
         if (client == null) {
             assert adminClient == null;
             assert clusterHosts == null;
+            assert hasXPack == null;
+            assert nodeVersions == null;
             String cluster = System.getProperty("tests.rest.cluster");
             if (cluster == null) {
                 throw new RuntimeException("Must specify [tests.rest.cluster] system property with a comma delimited list of [host:port] "
@@ -157,10 +151,27 @@ public abstract class ESRestTestCase extends ESTestCase {
             logger.info("initializing REST clients against {}", clusterHosts);
             client = buildClient(restClientSettings(), clusterHosts.toArray(new HttpHost[clusterHosts.size()]));
             adminClient = buildClient(restAdminSettings(), clusterHosts.toArray(new HttpHost[clusterHosts.size()]));
+
+            hasXPack = false;
+            nodeVersions = new TreeSet<>();
+            Map<?, ?> response = entityAsMap(adminClient.performRequest(new Request("GET", "_nodes/plugins")));
+            Map<?, ?> nodes = (Map<?, ?>) response.get("nodes");
+            for (Map.Entry<?, ?> node : nodes.entrySet()) {
+                Map<?, ?> nodeInfo = (Map<?, ?>) node.getValue();
+                nodeVersions.add(Version.fromString(nodeInfo.get("version").toString()));
+                for (Object module: (List<?>) nodeInfo.get("modules")) {
+                    Map<?, ?> moduleInfo = (Map<?, ?>) module;
+                    if (moduleInfo.get("name").toString().startsWith("x-pack-")) {
+                        hasXPack = true;
+                    }
+                }
+            }
         }
         assert client != null;
         assert adminClient != null;
         assert clusterHosts != null;
+        assert hasXPack != null;
+        assert nodeVersions != null;
     }
 
     /**
@@ -190,6 +201,8 @@ public abstract class ESRestTestCase extends ESTestCase {
             clusterHosts = null;
             client = null;
             adminClient = null;
+            hasXPack = null;
+            nodeVersions = null;
         }
     }
 
@@ -276,8 +289,6 @@ public abstract class ESRestTestCase extends ESTestCase {
     }
 
     private void wipeCluster() throws Exception {
-        boolean hasXPack = hasXPack();
-
         if (preserveIndicesUponCompletion() == false) {
             // wipe indices
             try {
@@ -325,18 +336,13 @@ public abstract class ESRestTestCase extends ESTestCase {
             wipeClusterSettings();
         }
 
-        if (hasXPack && false == preserveRollupJobsUponCompletion()) {
-            try {
-                wipeRollupJobs();
-            } catch (ResponseException e) {
-                // Temporary work around for mixed clusters, some of which don't have rollup
-                if (    // The request was made to node without rollup
-                        e.getResponse().getStatusLine().getStatusCode() != 400
-                        // or the request was made to a node with rollup and it sent something to a node without it
-                        && false == e.getMessage().contains("No handler for action [cluster:monitor/xpack/rollup/get]")) {
-                    throw e;
-                }
-            }
+        /*
+         * Rollups were introduced in 6.3.0 so any cluster that contains older
+         * nodes won't be able to do *anything* with rollups, including cleanup.
+         */
+        if (hasXPack && nodeVersions.first().onOrAfter(Version.V_6_3_0)
+                && false == preserveRollupJobsUponCompletion()) {
+            wipeRollupJobs();
             waitForPendingRollupTasks();
         }
     }

